### PR TITLE
Make entire public API C-compatible

### PR DESCRIPTION
**Pull request by denis-ismailaj**

### CODE REVIEW

File: examples/zsl.cpp [hunk 1]
```diff
@@ -22,20 +22,20 @@ int main(int argc, char ** argv) {
     }
 
     // load the image
-    std::string img_path = params.image_paths[0];
+    const char *img_path = params.image_paths[0].c_str();
```
Comment (Green-Sky) [inline examples/zsl.cpp:25]: const auto&. c-api and c-like-code are different things.

**Review by Green-Sky — CHANGES_REQUESTED** [review]
looks ok overall

File: examples/zsl.cpp (hunk 1 above)
Comment (monatis) [reply examples/zsl.cpp]: Fixed.

**Comment by denis-ismailaj** [comment]
> One thing in my mind is, maybe we can accept clip_image_f32 * images, size_t n_images instead of the clip_image_f32_batch struct. It seems to be tedious to create that struct.

Either that or provide a function that produces the batch struct given a std::vector.

**Comment by monatis** [comment]
Yes, maybe a util function in the `common-clip` lib. I'll look into that.

**Pull request merged**